src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -124,20 +124,25 @@ public ITest Build(string assemblyName, IDictionary<string, object> options)
 
         private TestSuite Build(Assembly assembly, string assemblyPath, IDictionary<string, object> options)
         {
-            TestSuite testAssembly = null;
-
             try
             {
-                if (options.ContainsKey(FrameworkPackageSettings.DefaultTestNamePattern))
-                    TestNameGenerator.DefaultTestNamePattern = options[FrameworkPackageSettings.DefaultTestNamePattern] as string;
+                var testNamePattern = GetValueOrDefault(options, FrameworkPackageSettings.DefaultTestNamePattern) as string;
+                if (testNamePattern != null) TestNameGenerator.DefaultTestNamePattern = testNamePattern;
 
-                if (options.ContainsKey(FrameworkPackageSettings.TestParameters))
+                var testParametersDictionary = GetValueOrDefault(options, FrameworkPackageSettings.TestParametersDictionary) as IDictionary<string, string>;
+                if (testParametersDictionary != null)
+                {
+                    foreach (var parameter in testParametersDictionary)
+                        TestContext.Parameters.Add(parameter.Key, parameter.Value);
+                }
+                else
                 {
-                    string parameters = options[FrameworkPackageSettings.TestParameters] as string;
-                    if (!string.IsNullOrEmpty(parameters))
-                        foreach (string param in parameters.Split(new[] { ';' }))
+                    // Kept for backwards compatibility with old runners
+                    var testParametersString = GetValueOrDefault(options, FrameworkPackageSettings.TestParameters) as string;
+                    if (!string.IsNullOrEmpty(testParametersString))
+                        foreach (var param in testParametersString.Split(';'))
                         {
-                            int eq = param.IndexOf("=");
+                            var eq = param.IndexOf("=");
```

Comment:
Yes, ReSharper. The original code mixed var and non-var styles so I figured I'd pick one. Easy enough to undo.